internals/src/error.rs

```diff
@@ -26,3 +26,23 @@ macro_rules! write_err {
         }
     }
 }
+
+/// Trait used internally to assert that error types implement `Send` and `Sync`.
+#[doc(hidden)]
+pub trait AssertSendSync: Send + Sync {}
+
+/// Trait used internally to assert that error types implement `Debug`.
+#[doc(hidden)]
+pub trait AssertDebug: core::fmt::Debug {}
+
+/// Checks at compile time that type `$t` implements the traits we require for error types.
+#[macro_export]
+macro_rules! check_pub_error {
+    ($t:ty) => {
+        #[cfg(test)]
+        impl internals::error::AssertSendSync for $t {}
+        #[cfg(test)]
+        #[cfg(feature = "std")]
+        impl internals::error::AssertDebug for $t {}
```

Comment:
Why is `Debug` gated on `std`? Seems wrong, all error types should impl `Display` and `Debug` and, in case of `std`, `std::error::Error`.

Reply:
My bad, sorry for wasting review time.